graphql_compiler/macros/macro_edge/directives.py

```diff
@@ -36,9 +38,24 @@
     ],
 )
 
-
+# Directives reserved for macro edges
 MACRO_EDGE_DIRECTIVES = (
     MacroEdgeDirective,
     MacroEdgeDefinitionDirective,
     MacroEdgeTargetDirective,
 )
+
+# Directives required present in a macro edge definition
+DIRECTIVES_REQUIRED_IN_MACRO_EDGE_DEFINITION = frozenset({
+    MacroEdgeDefinitionDirective,
+    MacroEdgeTargetDirective
+})
+
+# Directives (excluding MACRO_EDGE_DIRECTIVES) tolerated without restrictions within a macro
+# edge definition
+OPTIONAL_DIRECTIVES_UNRESTRICTED_IN_MACRO_EDGE_DEFINITION = frozenset({
```

Comment:
I don't think the `OPTIONAL` is helpful here. How about `UNRESTRICTED_DIRECTIVES_IN_MACRO_EDGE_DEFINITION` plus the comment you wrote right above it?

Naming is hard. But it's also very important.

Reply:
- If we're working with macro edges and directives, we'll need three sets: all allowed, required for macro-edge, and non-macro-edge (optional) specific. Since we're working with sets, given two of these, we can find the third using either a union or difference. So we should implement two. Just by going over the functions we have now, the first two seem to be most useful. So I agree, the optional directives set isn't needed.
- I don't think an UNRESTRICTED directives set should stay here. How can we communicate FoldDirective is allowed within macro edges? I think the better solution is to create an allowed directives set, whether they have restrictions or not, and then manually check each directive for its necessary conditions (FoldDirectives, MacroEdgeDefinition, MacroEdgeTarget).